requirements.txt

```diff
@@ -11,9 +11,13 @@ s3path
 pybind11
 popmon
 varclushi
-pytest
 cython
 soupsieve
 fsspec
 s3fs
 findspark<2.0.0
+pyspark>=2.4.3,<=2.4.8
```

Comment:
This breaks compatibility with Spark 3.x, no?

Reply:
yes but i got this is requirement when i scanned the project using poetry, i think it has conflict with some libray we call for

Reply:
We just introduced compatibility with Spark 3.x, so that's not an option :-)

Which package has the issues? Would be good to know! The demos and unit tests run fine under several Spark versions, but of course that's an imperfect test.

Reply:
checked again poetry and the following shall be supported:
![image](https://user-images.githubusercontent.com/5924038/153193963-4dc774ff-4d3c-4298-a030-34086908edc9.png)

Reply:
@ionicsolutions  will update the requirements for pyspark to cover `3.2.1`

Reply:
My current understanding is that `pyspark` is installed through Spark and not via `pip`. As it needs to match whatever Spark version a user has installed, I don't think we should add it in `requirements.txt`.

Reply:
Have a look at https://github.com/anovos/anovos/blob/main/.github/workflows/demo.yml for example

Reply:
That's a good idea. We're working on the dependencies anyway, some have been fixed to arbitrary versions for unknown reasons. The only thing we know we have issues with are datapane #48 (there was a breaking change between 0.12.x and 0.13.x) and findspark #50 (which is just a test requirement and I don't know why 2.0.0 doesn't work for us).